src/Lean/PrettyPrinter/Delaborator/Builtins.lean

```diff
@@ -128,6 +128,26 @@ def delabAppExplicit : Delab := do
       pure (fnStx, argStxs.push argStx))
   Syntax.mkApp fnStx argStxs
 
+inductive ImplicitArgKind : Type where
+  | provide
+  | skip
+  | name (n : Name)
+
+def returnsPi (motive : Expr) : Bool := do
+  match motive with
+  | Expr.lam name d b _ => returnsPi b
+  | Expr.forallE .. => true
+  | _ => false
+
+def isDepFun (motive : Expr) : Bool := do
```

Comment:
The name is a bit misleading, it's more `isNonConstFun`...? Perhaps the use site could come with some examples. Also feel free to use `where` for helper functions that are probably not reusable.

Reply:
Yes, fixed.